include/revng/Pipeline/RegisterContainerFactory.h

```diff
@@ -56,4 +56,31 @@ class RegisterDefaultConstructibleContainer : Registry {
   void libraryInitialization() override {}
 };
 
+/// Instantiate a global object of this class
+/// for each default constructible container to be used as role you wish to
+/// register
+template<typename ContainerT>
+class RegisterDefaultConstructibleRole : Registry {
```

Comment:
Let's just have `RegisterRole`, we'll never have non default constructible roles.
It's just a list of strings, let's not make it more complex that it needs to be.

Reply:
mhh i think we will, in particular if a container must be configured with a type

Reply:
That can be a template argument. Why do you think we'll need a constructor?

Reply:
mistake, i meant a kind

Reply:
I'd expect a role to be a map from string to container, why do you envision the need for a kind?